grub-core/kern/term.c

```diff
@@ -138,6 +138,22 @@ grub_getkeystatus (void)
   return status;
 }
 
+int
+grub_key_is_interrupt (int key)
+{
+  /* ESC sometimes is the BIOS setup hotkey and may be hard to discover, also
+     check F8, which was the key to get the Windows bootmenu for a long time. */
+  if (key == GRUB_TERM_ESC || key == GRUB_TERM_KEY_F8)
```

Comment:
Something that I missed yesterday, in the upstream discussion for this patch someone mentioned that F4 may be a better option than F8 and the the maintainer seems to agrees. So I guess that in v2 you will change that and also add the SHIFT pressed option that wasn't in your original patch?

So I think that we should also use F4 downstream, to be consistent with what would be upstream grub2 behaviour after your patch and also so we can drop this patch from the package once yours land and we do a rebase.